torch/multiprocessing/reductions.py

```diff
@@ -88,14 +88,27 @@ def rebuild_storage_filename(cls, manager, handle, size):
     return storage._shared_decref()
 
 
-def rebuild_storage_cuda(cls, device, handle, size, offset, view_size):
+def rebuild_storage_cuda(cls, device, handle, size, offset, view_size, originating_pid):
```

Comment:
Does this handle a situation where process `A` sends a tensor to `B`, and then `B` sends it back to `A`? I think in that case `originating_pid` would point to that of `B`. Additionally, that's not a very robust mechanism, because pids behave effectively like Python's `id` - you're guaranteed that it's unique for as long as a process is alive, but it can be reused once it dies, so you can't be 100% sure if this message was sent by you or not.

Maybe a better fix would be to just always normalize the storages in the cache to point to the beginning of the data block?

Reply:
I think you're right. I wrote a script to do that: process A sends a tensor to B, then B sends it back to A, and it appears to deadlock...